packages/astro/src/@types/astro.ts

```diff
@@ -2162,6 +2163,56 @@ export interface AstroUserConfig {
 			 */
 			schema?: EnvSchema;
 		};
+
+		/**
+		 * @docs
+		 * @name experimental.serverIslands
+		 * @type {boolean}
+		 * @default `false`
+		 * @version 4.12.0
+		 * @description
+		 *
+		 * Enables experimental Server Island features.
+		 * Server Islands offer the ability to defer a component to render asynchronously after the page has already rendered.
+		 *
+		 * Server Islands must be used with either `hybrid` or `server` output.
+		 * 
+		 * Use the `server:defer` directive on any Astro component and the component will not render initially.
+		 *
+		 * ```astro "server:defer"
+		 * <Avatar server:defer />
+		 * ```
+		 *
+		 * The outer page will be rendered, either at build-time (`hybrid`) or at runtime (`server`) with the island content omitted and a `<script>` tag included in its place.
+		 *
+		 * When the page loads in the browser the script tag will replace itself with the the contents of the island by making a request.
+		 *
+		 * Your island code can look like any normal Astro component; there is no special API for it:
+		 *
+		 * ```astro
+		 * ---
+		 * import { getUser } from '../api';
+		 *
+		 * const user = await getUser(Astro.locals.userId);
+		 * ---
+		 * <img class="avatar" src={user.imageUrl}>
+		 * ```
+		 *
+		 * #### Fallback
+		 *
+		 * Since your component doesn't render with the rest of the page you'll likely want to add fallback content. This content will be displayed when the page first renders but before the island has loaded.
+		 * 
+		 * Fallback is used to show generic content in the place of the islands. This could be loading messages or placeholder content. It could be a generic avatar that you animate towards the user avatar when it loads.
```

Comment:
```suggestion
		 * Since your component will not render with the rest of the page, you may want to add generic content (e.g. a loading message) to temporarily show in its place. This content will be displayed when the page first renders but before the island has loaded.
		 * 
		 * Add placeholder content as a child of your Astro component with the `slot="fallback:` attribute. When your island content is available, the fallback content will be replaced.
		 *
		 * The example below displays a generic avatar as fallback content, then animates into a personalized avatar using view transitions:
```
I'm inferring that `slot="fallback"` should be used here?

Reply:
Yes! Could call out